src/common/util/src/os/lin/lin_mmap_object.cpp

```diff
@@ -97,12 +97,26 @@ class MapHolder {
     }
 };
 
+typedef ngraph::runtime::SharedBuffer<std::shared_ptr<MapHolder>> SharedBufferMapHolder;
+
 std::shared_ptr<ngraph::runtime::AlignedBuffer> load_mmap_object(const std::string& path) {
     auto holder = std::make_shared<MapHolder>();
     holder->set(path);
-    return std::make_shared<ngraph::runtime::SharedBuffer<std::shared_ptr<MapHolder>>>(holder->data(),
-                                                                                       holder->size(),
-                                                                                       holder);
+    return std::make_shared<SharedBufferMapHolder>(holder->data(), holder->size(), holder);
+}
+
+std::shared_ptr<ov::op::v0::Constant> get_mmap_constant(const ov::element::Type type,
+                                                        const ov::Shape shape,
+                                                        std::shared_ptr<ngraph::runtime::AlignedBuffer> parent,
+                                                        const size_t offset,
+                                                        const size_t size) {
+    auto ptr = parent.get();
+    auto shared = dynamic_cast<SharedBufferMapHolder*>(ptr);
+    auto holder = shared->get_shared_object();
+    return std::make_shared<ov::op::v0::Constant>(
+        type,
+        shape,
+        std::make_shared<SharedBufferMapHolder>(static_cast<char*>(parent->get_ptr(offset)), size, holder));
```

Comment:
I think that we should check that `size` is not out of bounds, otherwise, it looks risky